cumulusci/cli/cci.py

```diff
@@ -594,28 +594,32 @@ def _build_param(self, attribute, details):
     def get_command(self, ctx, name):
         config = load_config()
 
-        attributes = iter(
-            list(
-                getattr(
-                    config.project_config, "services__{0}__attributes".format(name)
-                ).items()
+        try:
+            attributes = iter(
+                list(
+                    getattr(
+                        config.project_config, "services__{0}__attributes".format(name)
+                    ).items()
+                )
             )
-        )
-        params = [self._build_param(attr, cnfg) for attr, cnfg in attributes]
-        params.append(click.Option(("--project",), is_flag=True))
-
-        def callback(project=False, *args, **kwargs):
-            serv_conf = dict(
-                (k, v) for k, v in list(kwargs.items()) if v != None
-            )  # remove None values
-            config.keychain.set_service(name, ServiceConfig(serv_conf), project)
-            if project:
-                click.echo("{0} is now configured for this project".format(name))
-            else:
-                click.echo("{0} is now configured for global use".format(name))
-
-        ret = click.Command(name, params=params, callback=callback)
-        return ret
+            params = [self._build_param(attr, cnfg) for attr, cnfg in attributes]
+            params.append(click.Option(("--project",), is_flag=True))
+
+            def callback(project=False, *args, **kwargs):
+                serv_conf = dict(
+                    (k, v) for k, v in list(kwargs.items()) if v != None
+                )  # remove None values
+                config.keychain.set_service(name, ServiceConfig(serv_conf), project)
+                if project:
+                    click.echo("{0} is now configured for this project".format(name))
+                else:
+                    click.echo("{0} is now configured for global use".format(name))
+
+            ret = click.Command(name, params=params, callback=callback)
+            return ret
+
+        except AttributeError:
+            click.echo(name + " doesn't exist")
```

Comment:
Could you move the try/except so that it only catches AttributeError around the lines that try to get the `"services__{0}__attributes".format(name)` attribute? It's best to be as targeted as possible with exception handling, especially for AttributeError which can be raised pretty much anywhere if an attribute is misspelled (we don't have static type checking to catch that at compile time over here in Python land, at least not by default).

Reply:
I would do this instead of the click.echo:

    raise click.UsageError("Sorry, I don't know about the '{}' service.".format(name))

That results in a bit cleaner output:

    % cci service connect asdf
    Checking the version!
    Error: Sorry, I don't know about the 'asdf' service.

The difference between using `name + " string"` and `"{} string".format(name)` is subtle and isn't critical here. I generally tend to use the format method because it works even when you're interpolating a variable that is not a string, whereas using addition would raise a TypeError if given a different type. And the format method stays more consistent with the rest of the codebase.

Reply:
I checked the test coverage using `coverage run $(which pytest); coverage report -m` and it looks like we could use a test for the new except clause. Would you prefer to have a go at adding one on your own or have me do it and let you review? The test can go below `test_service_connect` in `test_cci.py`.

Reply:
Thanks for the detailed feedback, @davisagli! I'll be happy to continue working on this myself and implement the suggested changes, if there's no hurry getting them in.

Reply:
@ceiroa great! There's no hurry.